### PR TITLE
Add content search configurable content

### PR DESCRIPTION
**Pull request by itkg-nanne**
Most of functionalities work. Three issue : error on refresh content search, refresh form on submit, properties mapped false recorder.
Temporary PR. 
@betd-rcarles 
[OO-FEATURE] add content search widget in configurable content block

### CODE REVIEW

**Comment by betd-rcarles** [comment]
errors in UNITTEST -> cf Travis

File: Backoffice/EventSubscriber/ContentSearchSubscriber.php [hunk 1]
```diff
@@ -18,43 +18,92 @@ class ContentSearchSubscriber implements EventSubscriberInterface
...
+                if ($data['keywords'] != '') {
+                    $condition = json_decode($this->transformer->reverseTransform($data['keywords']), true);
+                }
+                $choices = array_merge($choices, $this->getChoices($data['contentType'], $data['choiceType'], $condition));
+            }
+            if (array_key_exists('contentId', $data) && $data['contentId'] != '') {
+                $choices = array_merge($choices, $this->getChoice($data['contentId']));
             }
+        }
+        if(count($choices) > 0) {
```
Comment (alavieille) [inline Backoffice/EventSubscriber/ContentSearchSubscriber.php]: psr2

File: Backoffice/EventSubscriber/ContentSearchSubscriber.php [hunk 2]
```diff
@@ -18,43 +18,92 @@ class ContentSearchSubscriber implements EventSubscriberInterface
...
             $form->add('contentId', 'choice', array(
                 'label' => false,
-                'required' => false,
-                'choices' => $this->getChoices($data['contentType'], $data['choiceType'], $condition),
+                'empty_value' => '        ',
```
Comment (alavieille) [inline Backoffice/EventSubscriber/ContentSearchSubscriber.php]: why multiple spaces

File: Backoffice/EventSubscriber/ContentSearchSubscriber.php [hunk 3]
```diff
@@ -86,4 +136,15 @@ protected function getChoices($contentType, $choiceType, $condition)
         return $choices;
     }
 
+    /**
+     * @param string $contentId
+     */
+    protected function getChoice($contentId)
```
Comment (alavieille) [inline Backoffice/EventSubscriber/ContentSearchSubscriber.php]: phpdoc return

**Comment by alavieille** [comment]
:+1:

**Pull request merged**